### PR TITLE
Fix cmd.Run() which should handle error but didn't

### PR DESCRIPTION
**Pull request by anfernee**

### CODE REVIEW

File: exec/exec.go [hunk 1]
```diff
@@ -148,16 +143,19 @@ func (cmd *cmdWrapper) Stop() {
...
-		return x
+	if err == nil {
+		return nil
 	}
-	if ee, ok := err.(*osexec.Error); ok {
-		if ee.Err == osexec.ErrNotFound {
+
+	switch e := err.(type) {
+	case *osexec.ExitError:
+		return &ExitErrorWrapper{e}
```
Comment (thockin) [inline exec/exec.go]: We should probably change this to a non-pointer - it doesn't need top be heap, I think.

Comment (anfernee) [reply exec/exec.go]: I just make it minimal. I think adding `&` or not doesn't change where it is allocated. anyway, thanks for the quick review.

**Comment by thockin** [comment]
It forces it to be on the heap, as far as I can tell. And yeah, this is
fine for now, but maybe a followup? :)

**Pull request merged**